src/pywifes/wifes_wsol.py

```diff
@@ -1568,62 +1560,56 @@ def _fit_optical_model(
-                        rmse = numpy.sqrt(chisq / dof)
-                        print("RMSE", rmse)
-
-                    # Copy back fitted parameters into parinfo structure
-                    for i, v in enumerate(m.params):
-                        parinfo[i]["value"] = v
-                        # Report the ones we just changed
-                        if (parinfo[i]["fixed"] == 0) and verbose:
-                            print(i, parinfo[i]["value"])
-
-                    # Repeat the fit if we need more steps
-                    if (m.status == 5) and (fitcount < MAXFITS):
-                        print("mpfit needs more steps; repeating fit")
-                    else:
-                        fitdone = True
-
-                if doplot:
-                    pl = numpy.asarray(m.params)[: om.nparams]
-                    resid = om.errfunc(grating, pl, alphap, alls, ally, allx, allarcs)
-                    om.plotResid(title, allx, ally, allarcs, resid)
-                    # om.plotFunc(title,allx,ally,allarcs,om.fitfunc(grating, pl,alphap,alls,ally,allx))
+    for params in paramlist:
+      for r in range(1):
+        if (verbose):
+          print('Fitting for parameters',params)
+        # Fix all parameters
+        for p in parinfo:
+          p['fixed'] = 1
+        # Unfix only those we want to fit this time around
+        for i in params:
+          parinfo[i]['fixed'] = 0
+        # Do the fit
+        fitdone = False
+        fitcount = 0
+        # Maximum number of times to repeat the fit
+        MAXFITS = 1
+        while (not fitdone):
+          fitcount += 1
+          # Actually do the fit
+          m = mpfit(om.mpfitfunc, functkw=fa, parinfo=parinfo, iterfunct=None, ftol=FTOL)
+          # Report on it
+          if (verbose):
+            print('status = ', m.status)
+          if (m.status <= 0):
+             print('error message = ', m.errmsg)
+
+          if (verbose):
+            # Work out the RMSE
+            chisq = m.fnorm
+            dof=len(allx)-len(m.params)
+            rmse=numpy.sqrt(chisq/dof)
+            print("RMSE",rmse)
+
+          # Copy back fitted parameters into parinfo structure
+          for i,v in enumerate(m.params):
+            parinfo[i]['value'] = v
+            # Report the ones we just changed
+            if (parinfo[i]['fixed'] == 0) and verbose:
+              print(i, parinfo[i]['value'])
+
+          # Repeat the fit if we need more steps
+          if ((m.status == 5) and (fitcount < MAXFITS)):
+            print('mpfit needs more steps; repeating fit')
+          else:
+            fitdone = True
+
+            # if plot:
+            #   pl = numpy.asarray(m.params)[:om.nparams]
+            #   resid = om.errfunc(grating, pl, alphap, alls, ally, allx, allarcs)
+            #   om.plotResid(title,allx,ally,allarcs,resid)
+            #   om.plotFunc(title,allx,ally,allarcs,om.fitfunc(grating, pl,alphap,alls,ally,allx))

```

Comment:
Same for this piece od commented-out code. Remove it if not useful.

Reply:
Same here. I'll back to it soon.